Test: Run unit tests 50 times

mokagio opened the pull request:
See #15140

mokagio committed:
Set test repetition mode for WordPressAuthenticator as "retry on failure"

This aims to mitigate CI flakiness we've seen since merging #1500.

mokagio committed:
Test - Set `parallelism: 50` for unit tests

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Try running on Xcode 16.2

mokagio committed:
Make `TestBackgroundURLSessionDelegate` `final` to address warning

mokagio committed:
Set test repetition mode to "retry on failure" for the main suite

The WooCommerce `UnitTests.xctestplan` is the one that runs the
WordPressAuthenticator tests, not the one for WordPressAuthenticator
that was edited in bd8e98d696f6b7705850aa2b9b174e67608eee8f.

To enable retry on failure and hopefully address CI flakiness, we need
to modify the test plan that actually runs in CI.

mokagio committed:
Revert "Try running on Xcode 16.2"

This reverts commit 9b49e1f8a76ef594b4b49f9782ee5c56cd33e2e2.

mokagio committed:
Run dedicated WordPressAuthenticator tests in CI

mokagio committed:
Use "Automatically manage signing" in WordPressAuthenticator unit tests

This makes it so that code signing is not required when trying to run
the tests from the terminal with `xcodebuild` or Fastlane `scan`.

mokagio committed:
Try again full suite, but with iPhone 16

mokagio committed:
Disable WordPressAuthenticator tests as part of the main unit tests plan

This is because of the flakiness experienced in CI when running them as
part of the main unit tests test plan.

Instead, we'll run the WordPressAuthenticator unit tests in a dedicated
CI step, which https://github.com/woocommerce/woocommerce-ios/pull/15141
has shown to be far more reliable.
